util/src/main/java/com/redhat/lightblue/util/JsonUtils.java

```diff
@@ -109,6 +124,126 @@ public static JsonNode json(Reader reader, boolean systemPropertySubstitution) t
+            } else {
+                return JsonNodeFactory.instance.numberNode( ((Number)value).intValue());
+            }
+        } else if(value instanceof Boolean) {
+            return JsonNodeFactory.instance.booleanNode( ((Boolean)value).booleanValue());
+        } else if(value instanceof Date) {
+            return JsonNodeFactory.instance.textNode(Constants.getDateFormat().format((Date)value));
+        } else {
+            return JsonNodeFactory.instance.textNode(value.toString());
+        }            
+    }
+
+    /**
+     * Converts a json document to an object tree of maps/lists/values
+     *
+     * If json is a value, then the corresponding Java object is returned.
+     *
+     * If json is an array, a List is returned. Elements of the list are converted recursively
+     *
+     * If json is an object, a Map is returned.
+     */
+    public static Object fromJson(JsonNode json) {
+        if(json==null||json instanceof NullNode) {
+            return null;
+        } else if(json instanceof ObjectNode) {
+            return fromJson((ObjectNode)json);
+        } else if(json instanceof ArrayNode) {
+            return fromJson((ArrayNode)json);
+        } else {
+            return valueFromJson( (ValueNode)json);
+        }
+    }
+
+    private static Object fromJson(ObjectNode json) {
+        HashMap ret=new HashMap();
+        for(Iterator<Map.Entry<String,JsonNode>> itr=json.fields();itr.hasNext();) {
+            Map.Entry<String,JsonNode> entry=itr.next();
+            ret.put(entry.getKey(),fromJson(entry.getValue()));
+        }
+        return ret;
+    }
+
+    private static Object fromJson(ArrayNode json) {
+        ArrayList ret=new ArrayList(json.size());
+        for(Iterator<JsonNode> itr=json.elements();itr.hasNext();) {
+            ret.add(fromJson(itr.next()));
+        }
+        return ret;
+    }
+
+    /**
+     * Converts a Java object tree containing maps and collections to json
+     */
+    public static JsonNode toJson(Object obj) {
+        if(obj==null) {
+            return JsonNodeFactory.instance.nullNode();
+        } else if(obj instanceof Map) {
+            return toJson( (Map)obj );
+        } else if(obj instanceof Collection) {
+            return toJson( (Collection)obj);
+        } else {
+            return valueToJson(obj);
+        }
+    }
+
+    private static JsonNode toJson(Map obj) {
+        ObjectNode node=JsonNodeFactory.instance.objectNode();
+        for(Iterator<Map.Entry> itr=obj.entrySet().iterator();itr.hasNext();) {
+            Map.Entry entry=itr.next();
+            node.set((String)entry.getKey(),toJson(entry.getValue()));
+        }
+        return node;
+    }
+
+    private static JsonNode toJson(Collection obj) {
+        ArrayNode node=JsonNodeFactory.instance.arrayNode();
+        for(Object x:obj) {
+            node.add(toJson(x));
+        }
+        return node;
+    }
+
+    /**
```

Comment:
What is the reasoning for putting this logic in JsonUtils and not as a part of AnyType? Maybe the same question put differently, is there a re-use case for this?